elasticsearch/shield.sls

```diff
@@ -78,45 +78,120 @@ copy_shield_config:
     - require_in:
       - service: start_elasticsearch
 
-/etc/elasticsearch/server.crt:
+/etc/elasticsearch/ca:
+  file:
+    - recurse
+    - source: salt://elasticsearch/ca
+    - template: jinja
+    - user: root
+    - group: root
+    - file_mode: 644
+    - require:
+      - pkg: elasticsearch
+
+/etc/elasticsearch/ca/private/cakey.pem:
   file:
     - managed
     - user: root
     - group: root
-    - mode: 664
-    - contents_pillar: elasticsearch:encryption:certificate
+    - mode: 400
+    - makedirs: true
+    - contents_pillar: elasticsearch:encryption:ca_key
+    - require:
+      - file: /etc/elasticsearch/ca
 
-/etc/elasticsearch/server.key:
+/etc/elasticsearch/ca/certs/cacert.pem:
   file:
     - managed
     - user: root
     - group: root
-    - mode: 664
-    - contents_pillar: elasticsearch:encryption:private_key
+    - mode: 400
+    - makedirs: true
+    - contents_pillar: elasticsearch:encryption:ca_cert
+    - require:
+      - file: /etc/elasticsearch/ca
+
+# Delete before re-creating to ensure idempotency
+delete-truststore:
```

Comment:
hmm, so other nodes shouldn't have to re-trust the new cert created in the keystore, since the new cert is signed by the CA cert (which should be in the truststore on all other nodes).  

And for non-es nodes, yeah if the CA cert changes, the clients will need the new CA cert in their truststores.

(It should be noted though that if only the CA cert changes, ES doesn't get restarted here, it requires a manual restart after the fact)